### DISCUSSION TITLE
linux - app image not working mx linux

### DISCUSSION THREAD

**Discussion by pkarza** [OrcaSlicer/OrcaSlicer#2275] [Q&A]
I just can't get the appimage to work in mx linux (debian).  I set the permissions to executable - and still... nothing.

I hate windows - but if I can't get this to work, I guess I have to use it.

any ideas what is going on?

**Comment by Arcaudas** [comment]
I get the same in Kubuntu.
After installed webkit2gtk-driver OrcaSlicer works!

**Reply by pkarza** [reply]
no joy.  didn't work.

**Comment by deck55** [comment]
I am getting the same thing on mint 21.2. It used to launch fine but something has changed, and now it will not launch.

edit: spelling

**Comment by kevlec** [comment]
same thing on ubuntu 23.10, even reinstalling the libraries, it doesn't work, do you have a tip?

**Reply by deck55** [reply]
I deleted this file (/home/"your user"/.config/OrcaSlicer/OrcaSlicer.conf) and it launched again.

**Reply by pkarza** [reply]
didn't work for me - and I even installed webkit2gtk-driver

**Reply by deck55** [reply]
You try to launch in terminal to see what errors you get?

**Reply by kevlec** [reply]
Hi,
I have just tested and deleted the hidden folders of Orcaslicer, I have just tested beta 2, unfortunately nothing works.
However in the terminal I get this:
Do you have an idea ?

QSocketNotifier: Can only be used with threads started with QThread
/tmp/.mount_OrcaSlpH6BN5/bin/orca-slicer: error while loading shared libraries: libwebkit2gtk-4.0.so.37: cannot open shared object file: No such file or directory

**Reply by kevlec** [reply]
It's okay, I just found how to install the missing library:

Step 1 : sudo apt-get update -y
Step 2 : sudo apt-get install -y libwebkit2gtk-4.0-37

it works for me :-)

**Reply by pkarza** [reply]
well - this didn't work, either.

**Reply by kevlec** [reply]
run from the terminal and look at the error it gives you.
/home/XXXXX/YYYYYY/OrcaSlicer_Linux_V1.8.0-beta2_2586feaf44a242e4a8ba1969c22f9126.AppImage

**Reply by pkarza** [reply]
Here it is:
---------------------------------

[2023-11-10 15:16:32.559230] [0x00007f1c9b0a0ec0] [trace]   Initializing StaticPrintConfigs
add font of HarmonyOS_Sans_SC_Bold returns 1
add font of HarmonyOS_Sans_SC_Regular returns 1
add font of NanumGothic-Regular returns 1
add font of NanumGothic-Bold returns 1
loading network module at /home/phil/.config/OrcaSlicer/plugins/libbambu_networking.so
error, dlerror is /home/phil/.config/OrcaSlicer/plugins/libbambu_networking.so: cannot open shared object file: No such file or directory
after dlopen, network_module is (nil)

(orca-slicer:41024): GLib-GObject-WARNING **: 15:16:33.121: invalid cast from 'wxPizza' to 'GtkCellLayout'

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.121: gtk_cell_layout_get_cells: assertion 'GTK_IS_CELL_LAYOUT (cell_layout)' failed

(orca-slicer:41024): GLib-GObject-WARNING **: 15:16:33.122: invalid cast from 'wxPizza' to 'GtkCellLayout'

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.122: gtk_cell_layout_get_cells: assertion 'GTK_IS_CELL_LAYOUT (cell_layout)' failed

(orca-slicer:41024): GLib-GObject-WARNING **: 15:16:33.124: invalid cast from 'wxPizza' to 'GtkCellLayout'

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.124: gtk_cell_layout_get_cells: assertion 'GTK_IS_CELL_LAYOUT (cell_layout)' failed

(orca-slicer:41024): GLib-GObject-WARNING **: 15:16:33.125: invalid cast from 'wxPizza' to 'GtkCellLayout'

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.125: gtk_cell_layout_get_cells: assertion 'GTK_IS_CELL_LAYOUT (cell_layout)' failed

(orca-slicer:41024): GLib-GObject-WARNING **: 15:16:33.126: invalid cast from 'wxPizza' to 'GtkCellLayout'

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.126: gtk_cell_layout_get_cells: assertion 'GTK_IS_CELL_LAYOUT (cell_layout)' failed

(orca-slicer:41024): GLib-GObject-WARNING **: 15:16:33.128: invalid cast from 'wxPizza' to 'GtkCellLayout'

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.128: gtk_cell_layout_get_cells: assertion 'GTK_IS_CELL_LAYOUT (cell_layout)' failed

(orca-slicer:41024): GLib-GObject-WARNING **: 15:16:33.129: invalid cast from 'wxPizza' to 'GtkCellLayout'

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.129: gtk_cell_layout_get_cells: assertion 'GTK_IS_CELL_LAYOUT (cell_layout)' failed

(orca-slicer:41024): GLib-GObject-WARNING **: 15:16:33.130: invalid cast from 'wxPizza' to 'GtkCellLayout'

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.130: gtk_cell_layout_get_cells: assertion 'GTK_IS_CELL_LAYOUT (cell_layout)' failed

(orca-slicer:41024): GLib-GObject-WARNING **: 15:16:33.132: invalid cast from 'wxPizza' to 'GtkCellLayout'

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.132: gtk_cell_layout_get_cells: assertion 'GTK_IS_CELL_LAYOUT (cell_layout)' failed

(orca-slicer:41024): GLib-GObject-WARNING **: 15:16:33.133: invalid cast from 'wxPizza' to 'GtkCellLayout'

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.133: gtk_cell_layout_get_cells: assertion 'GTK_IS_CELL_LAYOUT (cell_layout)' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.250: gtk_window_resize: assertion 'height > 0' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.254: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.254: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.255: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.255: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.255: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.255: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.255: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.255: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.263: gtk_window_resize: assertion 'height > 0' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.264: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.264: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.375: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.375: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.389: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.389: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.398: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.398: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.406: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.406: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.416: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.416: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.426: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:41024): Gtk-CRITICAL **: 15:16:33.426: gtk_widget_set_size_request: assertion 'width >= -1' failed

**Reply by lasselars** [reply]
**/.config/OrcaSlicer/plugins/libbambu_networking.so**
I have the same error. Curious. Did you find any solution?

(I also tried on two different computers to compile but that fails at 89% after completed the 'dep_OCCT')

DISTRIB_DESCRIPTION="Linux Mint 21.2 Victoria"
UBUNTU_CODENAME=jammy
Linux se-lx-1022 5.15.0-88-generic #98-Ubuntu SMP

**Comment by zetabemol** [comment]
En mi caso pasaba lo mismo. También lo tengo instalado en Windows 10. Lo que hice fue copiar la carpeta de config de orca slicer (Windows10/Users/User/AppData/Roaming/orcaslicer) de windows 10, y reemplazar la misma carpeta en linux (/home/user/.config/orcaslicer). De esa manera funcionó

**Comment by EllaFoxo** [comment]
Can reproduce, Arch Linux via the AppImage.

**Comment by Nathan22211** [comment]
Same here. I have my console output in an issue I made. Seems to be related to a breaking change in wxwidgets or something.

**Comment by kevlec** [comment]
Hello,
with V2.0.0, have a new error :
QSocketNotifier: Can only be used with threads started with QThread
/tmp/.mount_OrcaSl5PoX4L/bin/orca-slicer: error while loading shared libraries: libwebkit2gtk-4.0.so.37: cannot open shared object file: No such file or directory
